### ISSUE TITLE
dreamplug: No /sbin/init in root fs

### ISSUE THREAD

**Issue by jvalleroy** [freedombox/freedom-maker#74]
About 1 month ago I was able to build a working dreamplug image (for https://github.com/freedombox/freedom-maker/pull/69). But now when I build an image it's not able to boot.
```
U-Boot 2014.10+dfsg1-5 (Apr 07 2015 - 21:53:14)
Marvell-DreamPlug

SoC:   Kirkwood 88F6281_A1
DRAM:  512 MiB
WARNING: Caches not enabled
SF: Detected MX25L1605D with page size 256 Bytes, erase size 64 KiB, total 2 MiB
In:    serial
Out:   serial
Err:   serial
Net:   egiga0, egiga1
88E1116 Initialized on egiga0
88E1116 Initialized on egiga1
Warning: Your board does not use generic board. Please read
doc/README.generic-board and take action. Boards not
upgraded by the late 2014 may break or be removed.
Hit any key to stop autoboot:  3  2  1  0 
(Re)start USB...
USB0:   USB EHCI 1.00
scanning bus 0 for devices... ERROR: Invalid USB EP length (9)
ERROR: Invalid USB EP length (9)
8 USB Device(s) found
       scanning usb for storage devices... Device NOT ready
   Request Sense returned 02 3A 00
3 Storage Device(s) found
reading uImage
2032579 bytes read in 113 ms (17.2 MiB/s)
reading uInitrd
11175738 bytes read in 528 ms (20.2 MiB/s)
## Booting kernel from Legacy Image at 06400000 ...
   Image Name:   Debian kernel 4.6.0-1-marvell
   Created:      2016-07-02  13:00:49 UTC
   Image Type:   ARM Linux Kernel Image (uncompressed)
   Data Size:    2032515 Bytes = 1.9 MiB
   Load Address: 00008000
   Entry Point:  00008000
   Verifying Checksum ... OK
## Loading init Ramdisk from Legacy Image at 06900000 ...
   Image Name:   Debian ramdisk 4.6.0-1-marvell
   Created:      2016-07-02  13:00:49 UTC
   Image Type:   ARM Linux RAMDisk Image (gzip compressed)
   Data Size:    11175674 Bytes = 10.7 MiB
   Load Address: 00000000
   Entry Point:  00000000
   Verifying Checksum ... OK
   Loading Kernel Image ... OK

Starting kernel ...

Uncompressing Linux... done, booting the kernel.
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 4.6.0-1-marvell (debian-kernel@lists.debian.org) (gcc version 5.4.0 20160609 (Debian 5.4.0-4) ) #1 Debian 4.6.2-2 (2016-06-25)
[    0.000000] CPU: Feroceon 88FR131 [56251311] revision 1 (ARMv5TE), cr=0005397f
[    0.000000] CPU: VIVT data cache, VIVT instruction cache
[    0.000000] Machine model: Globalscale Technologies Dreamplug
[    0.000000] Memory policy: Data cache writeback
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 130048
[    0.000000] Kernel command line: console=ttyS0,115200 root=/dev/sda2 rootdelay=10
[    0.000000] PID hash table entries: 2048 (order: 1, 8192 bytes)
[    0.000000] Dentry cache hash table entries: 65536 (order: 6, 262144 bytes)
[    0.000000] Inode-cache hash table entries: 32768 (order: 5, 131072 bytes)
[    0.000000] Memory: 502608K/524288K available (3773K kernel code, 367K rwdata, 1416K rodata, 288K init, 289K bss, 21680K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] Virtual kernel memory layout:
[    0.000000]     vector  : 0xffff0000 - 0xffff1000   (   4 kB)
[    0.000000]     fixmap  : 0xffc00000 - 0xfff00000   (3072 kB)
[    0.000000]     vmalloc : 0xe0800000 - 0xff800000   ( 496 MB)
[    0.000000]     lowmem  : 0xc0000000 - 0xe0000000   ( 512 MB)
[    0.000000]     pkmap   : 0xbfe00000 - 0xc0000000   (   2 MB)
[    0.000000]     modules : 0xbf000000 - 0xbfe00000   (  14 MB)
[    0.000000]       .text : 0xc0008000 - 0xc0519924   (5191 kB)
[    0.000000]       .init : 0xc051a000 - 0xc0562000   ( 288 kB)
[    0.000000]       .data : 0xc0562000 - 0xc05bdcd0   ( 368 kB)
[    0.000000]        .bss : 0xc05bdcd0 - 0xc0606490   ( 290 kB)
[    0.000000] NR_IRQS:16 nr_irqs:16 16
[    0.000000] clocksource: orion_clocksource: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 9556302233 ns
[    0.000008] sched_clock: 32 bits at 200MHz, resolution 5ns, wraps every 10737418237ns
[    0.000323] Console: colour dummy device 80x30
[    0.000351] Calibrating delay loop... 1185.79 BogoMIPS (lpj=2371584)
[    0.024030] pid_max: default: 32768 minimum: 301
[    0.024160] Security Framework initialized
[    0.024177] Yama: disabled by default; enable with sysctl kernel.yama.*
[    0.024274] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.024293] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.025161] CPU: Testing write buffer coherency: ok
[    0.025276] ftrace: allocating 16337 entries in 32 pages
[    0.050619] Setting up static identity map for 0x8200 - 0x823c
[    0.050898] mvebu-soc-id: MVEBU SoC ID=0x6281, Rev=0x3
[    0.053665] devtmpfs: initialized
[    0.057874] VFP support v0.3: not present
[    0.058104] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.058420] pinctrl core: initialized pinctrl subsystem
[    0.059742] NET: Registered protocol family 16
[    0.060179] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.061128] cpuidle: using governor ladder
[    0.061153] cpuidle: using governor menu
[    0.061486] Feroceon L2: Enabling L2
[    0.061529] Feroceon L2: Cache support initialised.
[    0.061802] [Firmware Info]: /ocp@f1000000/ethernet-controller@72000/ethernet0-port@0: local-mac-address is not set
[    0.061898] [Firmware Info]: /ocp@f1000000/ethernet-controller@76000/ethernet1-port@0: local-mac-address is not set
[    0.065517] No ATAGs?
[    0.070712] clocksource: Switched to clocksource orion_clocksource
[    0.083546] VFS: Disk quotas dquot_6.6.0
[    0.083613] VFS: Dquot-cache hash table entries: 1024 (order 0, 4096 bytes)
[    0.091612] NET: Registered protocol family 2
[    0.092407] TCP established hash table entries: 4096 (order: 2, 16384 bytes)
[    0.092466] TCP bind hash table entries: 4096 (order: 2, 16384 bytes)
[    0.092516] TCP: Hash tables configured (established 4096 bind 4096)
[    0.092580] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.092602] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.092797] NET: Registered protocol family 1
[    0.093070] Unpacking initramfs...
[    1.065097] Freeing initrd memory: 10916K (c6900000 - c73a9000)
[    1.065907] futex hash table entries: 256 (order: -1, 3072 bytes)
[    1.065980] audit: initializing netlink subsys (disabled)
[    1.066048] audit: type=2000 audit(1.056:1): initialized
[    1.066499] workingset: timestamp_bits=28 max_order=17 bucket_order=0
[    1.066637] zbud: loaded
[    1.068552] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 251)
[    1.068704] io scheduler noop registered
[    1.068755] io scheduler cfq registered (default)
[    1.069925] kirkwood-pinctrl f1010000.pin-controller: registered pinctrl driver
[    1.070820] irq: Cannot allocate irq_descs @ IRQ35, assuming pre-allocated
[    1.071306] irq: Cannot allocate irq_descs @ IRQ67, assuming pre-allocated
[    1.071817] mv_xor f1060800.xor: Marvell shared XOR driver
[    1.091886] mv_xor f1060800.xor: Marvell XOR (Registers Mode): ( xor cpy intr )
[    1.092065] mv_xor f1060900.xor: Marvell shared XOR driver
[    1.111871] mv_xor f1060900.xor: Marvell XOR (Registers Mode): ( xor cpy intr )
[    1.112249] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    1.113278] console [ttyS0] disabled
[    1.113333] f1012000.serial: ttyS0 at MMIO 0xf1012000 (irq = 26, base_baud = 12500000) is a 16550A
[    1.597355] console [ttyS0] enabled
[    1.603144] mousedev: PS/2 mouse device common for all mice
[    1.609259] rtc-mv f1010300.rtc: rtc core: registered f1010300.rtc as rtc0
[    1.616396] i2c /dev entries driver
[    1.621297] ledtrig-cpu: registered to indicate activity on CPUs
[    1.627833] NET: Registered protocol family 17
[    1.632929] registered taskstats version 1
[    1.637179] zswap: loaded using pool lzo/zbud
[    1.642057] rtc-mv f1010300.rtc: setting system clock to 2013-01-01 10:25:25 UTC (1357035925)
[    1.652719] Freeing unused kernel memory: 288K (c051a000 - c0562000)
[    1.659132] This architecture does not have kernel memory protection.
Loading, please wait...
starting version 230
[    1.729628] random: systemd-udevd urandom read with 5 bits of entropy available
[    1.953365] usbcore: registered new interface driver usbfs
[    1.970367] libphy: orion_mdio_bus: probed
[    2.002899] usbcore: registered new interface driver hub
[    2.018556] SCSI subsystem initialized
[    2.038548] usbcore: registered new device driver usb
[    2.062066] mv643xx_eth: MV-643xx 10/100/1000 ethernet driver version 1.4
[    2.081521] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    2.114440] ehci-orion: EHCI orion driver
[    2.128332] sata_mv f1080000.sata: slots 32 ports 1
[    2.156489] scsi host0: sata_mv
[    2.161846] ata1: SATA max UDMA/133 irq 33
[    2.170740] mmc0: new high speed SDIO card at address 0001
[    2.183437] orion-ehci f1050000.ehci: EHCI Host Controller
[    2.191563] mv643xx_eth_port mv643xx_eth_port.0 eth0: port 0 with MAC address f0:ad:4d:01:34:e7
[    2.206631] orion-ehci f1050000.ehci: new USB bus registered, assigned bus number 1
[    2.232604] orion-ehci f1050000.ehci: irq 30, io mem 0xf1050000
[    2.262778] orion-ehci f1050000.ehci: USB 2.0 started, EHCI 1.00
[    2.279024] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002
[    2.285875] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.293144] usb usb1: Product: EHCI Host Controller
[    2.298059] usb usb1: Manufacturer: Linux 4.6.0-1-marvell ehci_hcd
[    2.304283] usb usb1: SerialNumber: f1050000.ehci
[    2.309749] mv643xx_eth_port mv643xx_eth_port.1 eth1: port 0 with MAC address f0:ad:4d:01:34:e8
[    2.388639] hub 1-0:1.0: USB hub found
[    2.402049] hub 1-0:1.0: 1 port detected
[    2.477229] ata1: SATA link down (SStatus 0 SControl F300)
Begin: Loading essential drivers ... done.
[    2.722741] usb 1-1: new high-speed USB device number 2 using orion-ehci
[    2.859479] usb 1-1: New USB device found, idVendor=1a40, idProduct=0101
[    2.866234] usb 1-1: New USB device strings: Mfr=0, Product=1, SerialNumber=0
[    2.873424] usb 1-1: Product: USB 2.0 Hub
[    2.881357] hub 1-1:1.0: USB hub found
[    2.888126] hub 1-1:1.0: 4 ports detected
[    3.162739] usb 1-1.1: new high-speed USB device number 3 using orion-ehci
[    3.259480] usb 1-1.1: New USB device found, idVendor=1a40, idProduct=0101
[    3.266414] usb 1-1.1: New USB device strings: Mfr=0, Product=1, SerialNumber=0
[    3.273787] usb 1-1.1: Product: USB 2.0 Hub
[    3.281834] hub 1-1.1:1.0: USB hub found
[    3.288754] hub 1-1.1:1.0: 4 ports detected
[    3.370739] usb 1-1.2: new high-speed USB device number 4 using orion-ehci
[    3.467728] usb 1-1.2: New USB device found, idVendor=0781, idProduct=5530
[    3.474655] usb 1-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    3.482019] usb 1-1.2: Product: Cruzer
[    3.485799] usb 1-1.2: Manufacturer: SanDisk
[    3.490098] usb 1-1.2: SerialNumber: 20052242831D84E016B7
[    3.512185] usb-storage 1-1.2:1.0: USB Mass Storage device detected
[    3.520974] scsi host1: usb-storage 1-1.2:1.0
[    3.527779] usbcore: registered new interface driver usb-storage
[    3.537614] usbcore: registered new interface driver uas
[    3.574753] usb 1-1.1.1: new high-speed USB device number 5 using orion-ehci
[    3.678857] usb 1-1.1.1: New USB device found, idVendor=05e3, idProduct=0723
[    3.685959] usb 1-1.1.1: New USB device strings: Mfr=3, Product=4, SerialNumber=0
[    3.693500] usb 1-1.1.1: Product: USB Storage
[    3.697889] usb 1-1.1.1: Manufacturer: Generic 
[    3.705850] usb-storage 1-1.1.1:1.0: USB Mass Storage device detected
[    3.715711] usb-storage 1-1.1.1:1.0: Quirks match for vid 05e3 pid 0723: 8000
[    3.723011] scsi host2: usb-storage 1-1.1.1:1.0
[    3.758750] usb 1-1.3: new high-speed USB device number 6 using orion-ehci
[    3.875729] usb 1-1.3: New USB device found, idVendor=0cf3, idProduct=9271
[    3.882656] usb 1-1.3: New USB device strings: Mfr=16, Product=32, SerialNumber=48
[    3.890288] usb 1-1.3: Product: UB93
[    3.893896] usb 1-1.3: Manufacturer: ATHEROS
[    3.898216] usb 1-1.3: SerialNumber: 12345
[    3.954752] usb 1-1.1.2: new high-speed USB device number 7 using orion-ehci
[    4.058857] usb 1-1.1.2: New USB device found, idVendor=05e3, idProduct=0723
[    4.065959] usb 1-1.1.2: New USB device strings: Mfr=3, Product=4, SerialNumber=0
[    4.073508] usb 1-1.1.2: Product: USB Storage
[    4.077898] usb 1-1.1.2: Manufacturer: Generic 
[    4.085973] usb-storage 1-1.1.2:1.0: USB Mass Storage device detected
[    4.095730] usb-storage 1-1.1.2:1.0: Quirks match for vid 05e3 pid 0723: 8000
[    4.103030] scsi host3: usb-storage 1-1.1.2:1.0
[    4.138750] usb 1-1.4: new full-speed USB device number 8 using orion-ehci
[    4.236105] usb 1-1.4: New USB device found, idVendor=0d8c, idProduct=000c
[    4.243037] usb 1-1.4: New USB device strings: Mfr=0, Product=1, SerialNumber=0
[    4.250409] usb 1-1.4: Product: C-Media USB Headphone Set  
[    4.287503] hidraw: raw HID events driver (C) Jiri Kosina
[    4.299843] usbcore: registered new interface driver usbhid
[    4.305479] usbhid: USB HID core driver
[    4.315833] input: C-Media USB Headphone Set   as /devices/platform/ocp@f1000000/f1050000.ehci/usb1/1-1/1-1.4/1-1.4:1.3/0003:0D8C:000C.0001/input/input0
[    4.383640] hid-generic 0003:0D8C:000C.0001: input,hidraw0: USB HID v1.00 Device [C-Media USB Headphone Set  ] on usb-f1050000.ehci-1.4/input3
[    4.523947] scsi 1:0:0:0: Direct-Access     SanDisk  Cruzer           1.03 PQ: 0 ANSI: 2
[    4.546827] sd 1:0:0:0: [sda] 31266816 512-byte logical blocks: (16.0 GB/14.9 GiB)
[    4.562905] sd 1:0:0:0: [sda] Write Protect is off
[    4.568509] sd 1:0:0:0: [sda] No Caching mode page found
[    4.573873] sd 1:0:0:0: [sda] Assuming drive cache: write through
[    4.586390]  sda: sda1 sda2
[    4.594771] sd 1:0:0:0: [sda] Attached SCSI removable disk
[    4.730563] scsi 2:0:0:0: Direct-Access     Generic  STORAGE DEVICE   9451 PQ: 0 ANSI: 0
[    4.915899] sd 2:0:0:0: [sdb] 7744512 512-byte logical blocks: (3.97 GB/3.69 GiB)
[    4.924505] sd 2:0:0:0: [sdb] Write Protect is off
[    4.930379] sd 2:0:0:0: [sdb] No Caching mode page found
[    4.935736] sd 2:0:0:0: [sdb] Assuming drive cache: write through
[    4.956780]  sdb: sdb1 sdb2 sdb3 < sdb5 >
[    4.970663] sd 2:0:0:0: [sdb] Attached SCSI removable disk
[    5.108567] scsi 3:0:0:0: Direct-Access     Generic  STORAGE DEVICE   9451 PQ: 0 ANSI: 0
[    5.124425] sd 3:0:0:0: [sdc] Attached SCSI removable disk
Begin: Running /scripts/init-premount ... done.
Begin: Mounting root file system ... Begin: Running /scripts/local-top ... done.
Begin: Running /scripts/local-premount ... [   12.579041] raid6: int32x1  gen()    81 MB/s
[   12.651060] raid6: int32x1  xor()    59 MB/s
[   12.722735] raid6: int32x2  gen()   123 MB/s
[   12.794791] raid6: int32x2  xor()    79 MB/s
[   12.867103] raid6: int32x4  gen()   123 MB/s
[   12.938993] raid6: int32x4  xor()    86 MB/s
[   13.011082] raid6: int32x8  gen()   117 MB/s
[   13.083103] raid6: int32x8  xor()    79 MB/s
[   13.087410] raid6: using algorithm int32x2 gen() 123 MB/s
[   13.092848] raid6: .... xor() 79 MB/s, rmw enabled
[   13.097668] raid6: using intx1 recovery algorithm
[   13.110642] xor: measuring software checksum speed
[   13.154718]    arm4regs  :  1082.000 MB/sec
[   13.198725]    8regs     :   650.000 MB/sec
[   13.242721]    32regs    :   900.000 MB/sec
[   13.246931] xor: using function: arm4regs (1082.000 MB/sec)
[   13.297738] Btrfs loaded
Scanning for Btrfs filesystems
[   13.469277] random: nonblocking pool is initialized
[   13.948036] BTRFS: device fsid da36507b-8b69-4a18-ac58-d4296a86e3eb devid 1 transid 1564 /dev/sda2
done.
Begin: Will now check root file system ... fsck from util-linux 2.28
[/bin/fsck.btrfs (1) -- /dev/sda2] fsck.btrfs -a /dev/sda2 
done.
[   14.096289] BTRFS info (device sda2): disk space caching is enabled
[   14.102631] BTRFS: has skinny extents
done.
Begin: Running /scripts/local-bottom ... done.
Begin: Running /scripts/init-bottom ... done.
run-init: /sbin/init: No such file or directory
Target filesystem doesn't have requested /sbin/init.
run-init: /sbin/init: No such file or directory
run-init: /etc/init: Permission denied
run-init: /bin/init: No such file or directory
/bin/sh: 0: can't access tty; job control turned off
# mount
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
udev on /dev type devtmpfs (rw,nosuid,relatime,size=251304k,nr_inodes=62826,mode=755)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,noexec,relatime,size=51384k,mode=755)
/dev/sda2 on / type btrfs (ro,relatime,space_cache,subvolid=5,subvol=/)
# ls /sbin
agetty		  fsck.cramfs	     iw			partprobe
badblocks	  fsck.ext2	     iwconfig		pivot_root
blkdeactivate	  fsck.ext3	     iwevent		plipconfig
blkdiscard	  fsck.ext4	     iwgetid		rarp
blkid		  fsck.ext4dev	     iwlist		raw
blockdev	  fsck.fat	     iwpriv		regdbdump
brctl		  fsck.minix	     iwspy		resize2fs
bridge		  fsck.msdos	     killall5		resolvconf
capsh		  fsck.nfs	     ldconfig		rmmod
cfdisk		  fsck.vfat	     logsave		route
cgmanager	  fsfreeze	     losetup		rtacct
cgproxy		  fstab-decode	     lsmod		rtmon
chcpu		  fstrim	     mii-tool		runuser
crda		  getcap	     mkdosfs		setcap
ctrlaltdel	  getpcaps	     mke2fs		sfdisk
debugfs		  getty		     mkfs		shadowconfig
depmod		  hwclock	     mkfs.bfs		slattach
dhclient	  ifconfig	     mkfs.cramfs	start-stop-daemon
dhclient-script   ifdown	     mkfs.ext2		sulogin
dmsetup		  ifquery	     mkfs.ext3		swaplabel
dmstats		  ifup		     mkfs.ext4		swapoff
dosfsck		  insmod	     mkfs.ext4dev	swapon
dosfslabel	  insserv	     mkfs.fat		switch_root
dumpe2fs	  installkernel      mkfs.minix		sysctl
e2fsck		  ip		     mkfs.msdos		tc
e2image		  ip6tables	     mkfs.vfat		tune2fs
e2label		  ip6tables-restore  mkhomedir_helper	udevadm
e2undo		  ip6tables-save     mkswap		unix_chkpwd
ebtables	  ipmaddr	     modinfo		unix_update
ebtables-restore  ipset		     modprobe		wipefs
ebtables-save	  iptables	     nameif		wpa_action
fatlabel	  iptables-restore   pam-tmpdir-helper	wpa_cli
fdisk		  iptables-save      pam_tally		wpa_supplicant
findfs		  iptunnel	     pam_tally2		xtables-multi
fsck		  isosize	     parted		zramctl
```

**Comment by jvalleroy** [comment]
After creating the symlink /sbin/init -> /lib/systemd/systemd, I'm able to boot again.

**Comment by jvalleroy** [comment]
Fixed by merging #79.
